quisp/modules/Application/Application.h

```diff
@@ -20,28 +20,27 @@ namespace modules {
  *
  *  \brief Application
  */
-class Application : public cSimpleModule {
- private:
+class Application : public IApplication {
+ public:
+  Application();
+  ~Application() {}
+
+ protected:
   int my_address;
 
-  int *other_end_node_addresses;
-  int num_of_other_end_nodes;
+  std::vector<int> other_end_node_addresses;
   bool is_e2e_connection;
   int number_of_resources;
   int num_measure;
 
-  virtual void initialize() override;
```

Comment:
I think `virtual` maybe good to have for readability, just to know that it's a virtual function from base class. But if one is familiar with OMNeT++ you'd know them, maybe it's cleaner not to have it. I'm fine with either way.

Reply:
I don't intend to inherit `Application`, so I removed the `virtual` specifier.

the `virtual` specifier generates a vtable (virtual method table) for the class in order to do "dynamic dispatch".
in this case, I introduced `IApplication` and all Application or similar modules are treated as `IApplication` or `cModule` like this
```cpp
IApplication *app = new Application;
// or
cModule *app = new Application;
```
so the runtime checks `cModule` or `IApplication`'s vtable when the `initialize` method called, and then
the runtime calls inherited class's method (`Application::initialize`).

Reply:
I was talking mainly about style guide. Since it'll be virtual even without declaring since it's a derived class overriding the same method.

I guess the `override` keyword is enough to distinguished (also add compiler check). 
After I checked style guide recommendation, it seems most people also don't use the `virtual` keyword.

I agree, It is cleaner without.

Reply:
ah, just I want to say behavior before style guide. 

we should not change the behavior for readability.
current our codebase has a lot of useless `virtual` keywords. but the useless `virtual` keyword increases memory footprint and might induce an unintended override bug.